.ci.sh

```diff
@@ -8,27 +8,93 @@ set -eo pipefail  # exit if non-zero error is encountered (even in a pipeline)
+set +euo pipefail; . ~/.bash_profile; set -euo pipefail;  # load sct
 
 echo *** CHECK PATH ***
-ls -lA bin  # Make sure all binaries and aliases are there
-source python/etc/profile.d/conda.sh  # to be able to call conda
+# Make sure all binaries and aliases are there
+for tool in \
+            isct_antsApplyTransforms \
+            isct_antsRegistration \
+            isct_antsSliceRegularizedRegistration \
+            isct_ComposeMultiTransform \
+            isct_convert_binary_to_trilinear \
+            isct_dice_coefficient \
+            isct_minc2volume-viewer \
+            isct_propseg \
+            isct_spine_detect \
+            isct_test_ants \
+            isct_train_svm \
+            sct_analyze_lesion \
+            sct_analyze_texture \
+            sct_apply_transfo \
+            sct_check_dependencies \
+            sct_compute_ernst_angle \
+            sct_compute_hausdorff_distance \
+            sct_compute_mscc \
+            sct_compute_mtr \
+            sct_compute_mtsat \
+            sct_compute_snr \
+            sct_convert \
+            sct_create_mask \
+            sct_crop_image \
+            sct_deepseg \
+            sct_deepseg_gm \
+            sct_deepseg_lesion \
+            sct_deepseg_sc \
+            sct_denoising_onlm \
+            sct_detect_pmj \
+            sct_dice_coefficient \
+            sct_dmri_compute_bvalue \
+            sct_dmri_compute_dti \
+            sct_dmri_concat_b0_and_dwi \
+            sct_dmri_concat_bvals \
+            sct_dmri_concat_bvecs \
+            sct_dmri_display_bvecs \
+            sct_dmri_moco \
+            sct_dmri_separate_b0_and_dwi \
+            sct_dmri_transpose_bvecs \
+            sct_download_data \
+            sct_extract_metric \
+            sct_flatten_sagittal \
+            sct_fmri_compute_tsnr \
+            sct_fmri_moco \
+            sct_get_centerline \
+            sct_image \
+            sct_label_utils \
+            sct_label_vertebrae \
+            sct_maths \
+            sct_merge_images \
+            sct_process_segmentation \
+            sct_propseg \
+            sct_qc \
+            sct_register_multimodal \
+            sct_register_to_template \
+            sct_resample \
+            sct_run_batch \
+            sct_smooth_spinalcord \
+            sct_straighten_spinalcord \
+            sct_testing \
+            sct_version \
+            sct_warp_template; \
+            do
+  command -v "$tool" || (echo "Missing tool: $tool"; exit 1)
+done
+
+source "$SCT_DIR/python/etc/profile.d/conda.sh"  # to be able to call conda
 conda activate venv_sct  # reactivate conda for the pip install below
 
 echo *** UNIT TESTS ***
 pip install coverage
 echo -ne "import coverage\ncov = coverage.process_startup()\n" > sitecustomize.py
 echo -ne "[run]\nconcurrency = multiprocessing\nparallel = True\n" > .coveragerc
 COVERAGE_PROCESS_START="$PWD/.coveragerc" COVERAGE_FILE="$PWD/.coverage" \
-  pytest
+  pytest -n 1
```

Comment:
i forgot, what is the "-n" again?

Reply:
ah, nevermind, it is explained in https://github.com/neuropoly/spinalcordtoolbox/issues/2957#issuecomment-729738495

maybe we want to add a comment so future readers of the code will know?

Reply:
I would have if I thought it would stay. I don't really want to commit this like this. We have that open issue about it and I'd be happier if we could fix that first!

Reply:
```suggestion
  # -n 1 sidesteps a concurrency bug; see https://github.com/neuropoly/spinalcordtoolbox/issues/2957#issuecomment-729738495
  # it should be removed once that bug is fixed
  pytest -n 1
```

Reply:
I'm leaving this open because I feel like commit this with the concurrency bug still in place is bad form. There's gotta be some way to squash it quick. I'm sure we're not the only ones setting up account-wide fixtures with pytest. Maybe there's a better hook to use than

https://github.com/neuropoly/spinalcordtoolbox/blob/d64af23b69dbc396322cdc963a90a899b73736e0/conftest.py#L28

Reply:
I think this *was* easy: according to https://stackoverflow.com/questions/17801300/how-to-run-a-method-before-all-tests-in-all-classes/17844938#17844938 the usual way is

```
+@pytest.fixture(scope="session", autouse=True)
+def testdata():
-def pytest_sessionstart():
```

The (admittedly old) info on https://stackoverflow.com/questions/12586489/pytest-are-pytest-sessionstart-and-pytest-sessionfinish-valid-hooks/12600154#12600154 says

> Sessionstart in particular is not much documented because the semantics differ if one is in the xdist/distribution case or not.

So I think that's the basic problem here. We used the wrong API by accident.

I'm testing this fix in 5e45b3339820aa01c827554f015a9684fcc25444. I'll have to try it a few times to see if it makes the errors go away.

Reply:
This is backed out now. I sorted it in #3152. I just want to tidy that up (with input) and then this should be good to go on top of it. And @Drulex and I talked and maybe we'll drop xdist entirely because it's been more trouble than it's worth.